frontend/apps/reader/modules/readersearch.lua

```diff
@@ -71,13 +82,58 @@ function ReaderSearch:onShowFulltextSearchInput()
     self.input_dialog:onShowKeyboard()
 end
 
-function ReaderSearch:onShowSearchDialog(text, direction)
+function ReaderSearch:onShowRegexSearchInput()
```

Comment:
(May be move the more complex onShowRegexSearchInput function after the simpler onShowSearchDialog.)